index.js

```diff
@@ -0,0 +1,88 @@
+//dependency imports
+const express = require("express");
+
+//in-app imports
+const db = require("./data/db");
+
+//init server
+const server = express();
+
+//middleware
+server.use(express.json());
+
+//endpoints
+server.get("/", (req, res) => {
+  res.json({ server: "listening" });
+});
+
+//get all dishes in database
+server.get("/api/dishes", (req, res) => {
+  db.getDishes()
+    .then(dishes => {
+      res.status(200).json(dishes);
+    })
+    .catch(err => res.status(500).json({ error: "Couldn't retrieve dishes." }));
+});
+
+//add dish to database
+server.post("/api/dishes", (req, res) => {
+  const dish = req.body;
+  if (!dish.name) {
+    res.status(400).json({ message: "You have to name your dish something." });
+  } else {
+    db.addDish(dish)
+      .then(id => res.status(201).json(id))
+      .catch(err => res.status(500).json({ error: "Couldn't add dish." }));
+  }
+});
+
+//get dish by id, along with it's related recipes
+server.get("/api/dishes/:id", (req, res) => {
+  const { id } = req.params;
+  db.getDish(id)
+    .then(dish => res.status(200).json(dish))
+    .catch(err => res.status(500).json({ error: "Could not retrieve dish." }));
```

Comment:
It is good practice to always contain catch blocks in their separate block elements with curly braces `{}` to avoid undhandled errors and weird issues in the future.

Reply:
thanks for pointing that out for me Lukasz! I really appreciate you taking the time to review some of my repos, I'm always happy to have extra eyes on my code to help me catch things like that